src/storage/src/sink/kafka.rs

```diff
@@ -421,8 +412,27 @@ impl KafkaSinkState {
 
         let retry_manager = Arc::new(Mutex::new(KafkaSinkSendRetryManager::new()));
 
-        let producer_context =
-            SinkProducerContext::new(Arc::clone(&metrics), Arc::clone(&retry_manager));
+        let (status_tx, mut status_rx) = mpsc::channel(16);
+
+        let producer_context = SinkProducerContext {
+            metrics: Arc::clone(&metrics),
+            healthchecker: status_tx,
+            retry_manager: Arc::clone(&retry_manager),
+        };
+
+        let healthchecker: Arc<Mutex<Option<Healthchecker>>> = Arc::new(Mutex::new(None));
+
+        {
+            let healthchecker = Arc::clone(&healthchecker);
+            task::spawn(|| "producer-error-report", async move {
+                while let Some(status) = status_rx.recv().await {
+                    if let Some(hc) = healthchecker.lock().await.as_mut() {
```

Comment:
should we maybe retry here when we don't yet have a healthchecker? Might not be worth it, though.

Reply:
I think I'd like to try and refactor so that we always have a healthchecker, now that writes are enabled, which might avoid the conditional here? Will try it out.